File: app/src/main/java/com/p1/uber/provides/NotificationProvider.java
package com.p1.uber.provides;

import com.p1.uber.modelos.FCMBody;
import com.p1.uber.modelos.FCMResponce;
import com.p1.uber.retrofit.IFCMApi;
import com.p1.uber.retrofit.RetrofitClient;

import retrofit2.Call;

public class NotificationProvider {

    private String url = "https://fcm.googleapis.com";



    public NotificationProvider() {

    }

    public Call<FCMResponce> sendNotification(FCMBody body){
        return RetrofitClient.getClientObject(url).create(IFCMApi.class).send(body);
    }
}

File: app/src/main/java/com/p1/uberfares/provides/NotificationProvider.kt
package com.p1.uberfares.provides

import com.p1.uberfares.modelos.FCMBody
import com.p1.uberfares.modelos.FCMResponce
import com.p1.uberfares.retrofit.IFCMApi
import com.p1.uberfares.retrofit.RetrofitClient.getClientObject
import retrofit2.Call

class NotificationProvider {
    private val url = "https://fcm.googleapis.com"
    fun sendNotification(body: FCMBody?): Call<FCMResponce?>? {
        return getClientObject(url).create(IFCMApi::class.java).send(body)
    }
}